indigo_api/data_migrations/__init__.py

```diff
@@ -94,3 +96,84 @@ def update_namespace(self, xml):
             1)
 
         return xml
+
+
+class AKNeId:
+    """ In a document's XML, translate existing `id` attributes to `eId`
+    and follow the new naming and numbering convention
+    e.g. "sec_2" instead of "section-2"
+    """
+    basic_replacements = {
+        # TODO: make sure all have been added
+        # TODO: add a test for each type
+        "chapter": (r"\bchapter-", "chp_"),
+        "part": (r"\bpart-", "part_"),
+        "subpart": (r"\bsubpart-", "subpart_"),
+        "section": (r"\bsection-", "sec_"),
+        "paragraph": (r"\bparagraph-", "para_"),
+        "article": (r"\barticle-", "article_"),
+        "table": (r"\btable(?=\d)", "table_"),
+        "blockList": (r"\blist(?=\d)", "list_"),
+    }
+
+    def update_xml(self, xml):
+        """ Update the xml,
+            - first by updating the values of `id` attributes as necessary,
+            - and then by replacing all `id`s with `eId`s
+        """
+        doc = Act(xml)
+        self.namespace = doc.namespace
+
+        # basic replacements, e.g. "chapter-" to "chap_"
+        for element, patterns in self.basic_replacements.items():
```

Comment:
You can be extra fun here:
```suggestion
        for element, (pattern, replacement) in self.basic_replacements.items():
```
and then not have to refer to patterns[0] etc.

Reply:
nice!